os_migrate/plugins/module_utils/user.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import openstack
+
+from ansible_collections.os_migrate.os_migrate.plugins.module_utils \
+    import const, reference, resource
+
+
+class User(resource.Resource):
+    resource_type = const.RES_TYPE_USER
+    sdk_class = openstack.identity.v3.user.User
+
+    info_from_sdk = [
+        'id',
+        'domain_id',
+        'default_project_id',
+    ]
+
+    params_from_sdk = [
+        'name',
+        'description',
+        'email',
+        'is_enabled',
+    ]
+
+    params_from_refs = [
+        'default_project_ref',
+    ]
+
+    sdk_params_from_refs = [
+        'default_project_id',
+    ]
+
+    @staticmethod
+    def _refs_from_sdk(conn, sdk_res):
+        refs = {}
+        refs['default_project_ref'] = reference.user_ref(
+            conn, sdk_res['default_project_id'])
```

Comment:
We can't use user_ref on a project id. We're looking up a user and passing it a project id - so that will not find any user :). We need to use project_ref function here.